### PR TITLE
refactor acs-engine to separate project

### PR DESCRIPTION
**Pull request by mjudeikis**
PTAL @kargakis 

cc: @openshift/sig-azure this is so all of us could have access in the isolated namespaces and go get rid of test images with confusing version (3.10 jobs are using 3.9 image https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/test_cases/azure_build_node_image_rhel_310.yml#L78 )

Fixes https://github.com/openshift/release/issues/951

### CODE REVIEW

File: projects/acs-engine/objects.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Group will have some elevated permissions to deploy/maintain cluster-operator:
+- apiVersion: user.openshift.io/v1
+  kind: Group
+  metadata:
+    name: azure-team
+  users:
+  - mjudeikis
+  - jim-minter
+  - kargakis
+  - pweil-
```
Comment (kargakis) [inline projects/acs-engine/objects.yaml]: add kwoodson

File: projects/acs-engine/test-image/3.9.0/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/deis/go-dev:v1.9.1
+FROM dummy
```
Comment (kargakis) [inline projects/acs-engine/test-image/3.9.0/Dockerfile]: No need for this change even if it's not overriden by the build, please revert

File: projects/acs-engine/test-image/3.10.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM dummy
```
Comment (kargakis) [inline projects/acs-engine/test-image/3.10.0/Dockerfile]: quay.io/deis/go-dev:v1.9.1

File: projects/acs-engine/objects.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    namespace: azure
+  groupNames:
+  - aos-azure
+  roleRef:
+    name: admin
+  subjects:
+  - kind: Group
+    name: aos-azure
+
+- kind: ImageStream
```
Comment (kargakis) [inline projects/acs-engine/objects.yaml]: Add this back with the builds and keep the policy resources here.

Comment (mjudeikis) [reply projects/acs-engine/objects.yaml]: If we create separate builds using the same template which produces different tag to same imageStream - it does not work.

Comment (kargakis) [reply projects/acs-engine/objects.yaml]: I think I am in favor of removing the templatization from the other template and keep explicit build config. In my experience it's much better to manage than an openshift template albeit more config.

**Comment by kargakis** [comment]
/lgtm

thanks MJ for working on this!

**Pull request merged**